src/jit/codegenlegacy.cpp

```diff
@@ -5426,6 +5426,9 @@ void CodeGen::genCodeForTreeLeaf_GT_JMP(GenTreePtr tree)
             offset += REGSIZE_BYTES;
         }
     }
+
+    // Lock the argument registers s.t. they are not trashed by code that runs before the jmp (e.g. GS cookie checks).
+    regSet.rsLockReg(fixedArgsMask);
```

Comment:
FWIW, we could probably just mark the argument registers as used rather than used and locked. I don't think that locking them will be an issue, but it is likely to be stronger than what we need.